src/tests/string/test_strings_format_string.f90

```diff
@@ -1,24 +1,82 @@
 program test_strings_format_string
-    use, non_intrinsic :: stdlib_strings, only: format_string
+    use stdlib_strings, only: format_string, starts_with
+    use stdlib_error, only: check
+    use stdlib_optval, only: optval
     implicit none
     print *, 'format_string(complex) : '
-        print *, format_string((1, 1))
-        print *, format_string((1, 1), '(F6.2)')
-        print *, format_string((1, 1), '(F6.2)'), format_string((2, 2), '(F7.3)')
+    call check_formatter(format_string((1, 1)), "(1.00000000,1.00000000)", &
+        & "Default formatter for complex number")
+    call check_formatter(format_string((1, 1), '(F6.2)'), "(  1.00,  1.00)", &
+        & "Formatter for complex number")
+    call check_formatter(format_string((-1, -1), '(F6.2)'), "( -1.00, -1.00)", &
+        & "Formatter for negative complex number")
+    call check_formatter(format_string((1, 1), '(SP,F6.2)'), "( +1.00, +1.00)", &
+        & "Formatter with sign control descriptor for complex number")
+    call check_formatter(format_string((1, 1), '(F6.2)')//format_string((2, 2), '(F7.3)'), &
+        & "(  1.00,  1.00)(  2.000,  2.000)", &
+        & "Multiple formatters for complex numbers")
     print *, 'format_string(integer) : '
-        print *, format_string(100)
-        print *, format_string(100, '(I6)')
-        print *, format_string(100, '(I6)'), format_string(1000, '(I7)')
+    call check_formatter(format_string(100), "100", &
+        & "Default formatter for integer number")
+    call check_formatter(format_string(100, '(I6)'), "   100", &
+        & "Formatter for integer number")
+    call check_formatter(format_string(100, '(I0.6)'), "000100", &
+        & "Formatter with zero padding for integer number")
+    call check_formatter(format_string(100, '(I6)')//format_string(1000, '(I7)'), &
+        & "   100   1000", &
+        & "Multiple formatters for integers")
+    call check_formatter(format_string(34, '(B8)'), "  100010", &
+        & "Binary formatter for integer number")
+    call check_formatter(format_string(34, '(O0.3)'), "042", &
+        & "Octal formatter with zero padding for integer number")
+    call check_formatter(format_string(34, '(Z3)'), " 22", &
+        & "Hexadecimal formatter for integer number")
```

Comment:
👍Good idea. The test cases for binary, octal and hexadecimal format are inspiring.